File: middleware/src/dsl/Message.java
package dsl;

import java.util.ArrayList;
import java.util.List;

public class Message {
	private String name;
	private List<Component> from = new ArrayList<Component>();
	private List<Component> to = new ArrayList<Component>();
	private double minFrequency;
	private double maxFrequency;
	private String head;
	private String body;
	
	public Message(String name) {
		this.name = name;
	}
	
	public Message(String name, Component fromSingle, Component toSingle) {
		this(name);
		this.from.add(fromSingle);
		this.to.add(toSingle);
	}
	
	public Message(String name, List<Component> fromMultiple, Component toSingle) {
		this(name);
		this.from.addAll(fromMultiple);
		this.to.add(toSingle);
	}
	
	public Message(String name, List<Component> fromMultiple, List<Component> toMultiple) {
		this(name);
		this.from.addAll(fromMultiple);
		this.to.addAll(toMultiple);
	}
	
	public Message(String name, Component fromSingle, List<Component> toMultiple) {
		this(name);
		this.from.add(fromSingle);
		this.to.addAll(toMultiple);
	}
	
	public boolean isTo(Component c) {
		return (this.to == c);
	}
	
	public boolean isFrom(Component c) {
		return (this.from == c);
	}
	
	public List<Component> getTo() {
		return to;
	}
	
	public List<Component> getFrom() {
		return from;
	}
	
	public String getName() {
		return name;
	}
}

File: middleware/dsl/mission/Message.py
class Message:
	name = ""
	minFrequency = 0
	maxFrequency = 0
	head = ""
	
	def __init__(self, id, gid, name, sender, receiver, data):
		self.name = name
		self.sender = sender
		self.receiver = receiver
		self.data = data
		self.ID = id
		self.goal_ID = gid
	
	def is_to(self, c):
		return (self.receiver == c)
	
	def is_from(self, c):
		return (self.sender == c)
	
	def get_to(self):
		return self.receiver
	
	def get_from(self):
		return self.sender
	
	def get_name(self):
		return self.name
